clang-tidy crashes when it encounters #embed in C code

I ran into the following crash trying to run clang-tidy on some code using C23 `#embed` to include a JSON schema file into a `const char []`. The shortened output from clang-tidy (the [full-output](https://github.com/user-attachments/files/16920084/clang-tidy19-embed-crash.txt) is attached just in case):
```
LLVM ERROR: Support for EmbedExpr is not implemented.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /usr/local/llvm19/bin/clang-tidy main.c -- -I/usr/local/include -fblocks
1.	<eof> parser at end of file
2.	While analyzing stack:
	#0 Calling get_schema at line 99
	#1 Calling main
3.	/home/crest/friendly-s6/main.c:1074:3: Error evaluating statement
4.	/home/crest/friendly-s6/main.c:1074:3: Error evaluating statement
```



Here's the trimmed down code to the function that triggers the error (with line numbers to match up against the crash report).
```
  1070 static inline ucl_object_t *_Nonnull
  1071 get_schema(void)
  1072 {
  1073         const unsigned char schema_json[] = {
  1074                 #embed "schema.json"
  1075         };
```

--- a/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp
+++ b/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp
@@ -1938,6 +1938,7 @@ void ExprEngine::Visit(const Stmt *S, ExplodedNode *Pred,
     case Stmt::CXXRewrittenBinaryOperatorClass:
     case Stmt::RequiresExprClass:
     case Expr::CXXParenListInitExprClass:
+    case Stmt::EmbedExprClass:
       // Fall through.
 
     // Cases we intentionally don't evaluate, since they don't need
@@ -2440,10 +2441,6 @@ void ExprEngine::Visit(const Stmt *S, ExplodedNode *Pred,
       Bldr.addNodes(Dst);
       break;
     }
-
-    case Stmt::EmbedExprClass:
-      llvm::report_fatal_error("Support for EmbedExpr is not implemented.");
-      break;
   }
 }
 

--- a/clang/test/Analysis/embed.c
+++ b/clang/test/Analysis/embed.c
@@ -1,0 +1,9 @@
+// RUN: %clang_analyze_cc1 -std=c23 -analyzer-checker=core,debug.ExprInspection -verify %s
+
+// expected-no-diagnostics
+
+int main() {
+    const unsigned char SelfBytes[] = {
+        #embed "embed.c"
+    };
+}

--- a/clang/test/Analysis/embed.c
+++ b/clang/test/Analysis/embed.c
@@ -1,9 +1,12 @@
 // RUN: %clang_analyze_cc1 -std=c23 -analyzer-checker=core,debug.ExprInspection -verify %s
 
-// expected-no-diagnostics
+void clang_analyzer_dump_ptr(const unsigned char *ptr);
+void clang_analyzer_dump(unsigned char val);
 
 int main() {
     const unsigned char SelfBytes[] = {
         #embed "embed.c"
     };
+    clang_analyzer_dump_ptr(SelfBytes); // expected-warning {{&Element{SelfBytes,0 S64b,unsigned char}}}
+    clang_analyzer_dump(SelfBytes[0]); // expected-warning {{Unknown}} FIXME: This should be the `/` character.
 }

